telemetry/telemetrybuffer.go

```diff
@@ -426,3 +420,47 @@ func getHostMetadata() (Metadata, error) {
 
 	return metareport.Metadata, err
 }
+
+// WaitForTelemetrySocket - Block still pipe/sock created or until max attempts retried
+func WaitForTelemetrySocket(maxAttempt int, waitTimeInMillisecs time.Duration) {
+	for attempt := 0; attempt < maxAttempt; attempt++ {
+		if SockExists() {
+			break
+		}
+
+		time.Sleep(waitTimeInMillisecs * time.Millisecond)
+	}
+}
+
+// StartTelemetryService - Kills if any telemetry service runs and start new telemetry service
+func StartTelemetryService(path string, args []string) error {
+	platform.KillProcessByName(TelemetryServiceProcessName)
+
+	log.Printf("[Telemetry] Starting telemetry service process :%v args:%v", path, args)
+
+	if err := common.StartProcess(path, args); err != nil {
+		log.Printf("[Telemetry] Failed to start telemetry service process :%v", err)
+		return err
+	}
+
+	log.Printf("[Telemetry] Telemetry service started")
+
+	return nil
+}
+
+// ReadConfigFile - Read telemetry config file and populate to structure
+func ReadConfigFile(filePath string) (TelemetryConfig, error) {
+	config := TelemetryConfig{}
+
+	b, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Printf("[Telemetry] Failed to read telemetry config: %v", err)
+		return config, err
```

Comment:
we should return nil config here, no need for empty

Reply:
nope..this is good instead of nil..return by value is better than returning pointers in golang

Reply:
I see that the config is really small so it's ok